smartsim/entity/ensemble.py

```diff
@@ -59,23 +59,182 @@ def __init__(
         max_permutations: int = -1,
         replicas: int = 1,
     ) -> None:
+        """Initialize an ``Ensemble`` of application instances
+
+        :param name: name of the ensemble
+        :param exe: executable to run
+        :param exe_args: executable arguments
+        :param exe_arg_parameters: parameters and values to be used when configuring entities
+        :param files: files to be copied, symlinked, and/or configured prior to
+                      execution
+        :param file_parameters: parameters and values to be used when configuring
+                                files
+        :param permutation_strategy: strategy to control how the param values are applied to the Ensemble
+        :param max_permutations: max parameter permutations to set for the ensemble
+        :param replicas: number of identical entities to create within an Ensemble
+        :return: and ``Ensemble`` instance
```

Comment:
nit: `__init__` doesn't actually return a new ensemble instance (that's the job of [`__new__`](https://docs.python.org/3/reference/datamodel.html#object.__new__)). Feel free to delete this line!
```suggestion
```

Reply:
removed